src/dfx/src/lib/identity/mod.rs

```diff
@@ -558,7 +559,7 @@ impl Identity {
         match Identity::wallet_canister_id(network, name)? {
             None => {
                 // If the network is not the IC, we ignore the error and create a new wallet for the identity.
-                if !network.is_ic || create {
+                if (!network.is_ic && std::env::var("DISABLE_AUTO_WALLET").is_err()) || create {
```

Comment:
I'm having a little trouble following this logic (to be fair, it was hard to follow before, too).

If this environment variable is controlling something only on mainnet, should we reflect that in the name?

I've submitted https://github.com/dfinity/sdk/pull/2466 , which removes the `create` parameter altogether, and may help here.

Reply:
> I've submitted https://github.com/dfinity/sdk/pull/2466 , which removes the create parameter altogether, and may help here.
Good change, but does not affect this.

This does only change the behaviour on non-mainnet (the `!is_ic` flag). The 'problem' is that on local (and test) networks dfx automatically creates a wallet whenever `get_or_create_wallet` is called. This is very convenient for `dfx start --background && dfx deploy` since the wallet is automatically created, but it makes it impossible to test the case where no wallet exists.

The introduced flag means that no wallet gets automatically created on non-mainnets. Otherwise it would not be possible to test the case where the user has no wallet configured yet.